src/TurboMqtt/Streams/MqttEncodingFlows.cs

```diff
@@ -34,7 +34,7 @@ public static class MqttEncodingFlows
             .Select(c => (c, MqttPacketSizeEstimator.EstimateMqtt3PacketSize(c)))
             .Via(new PacketSizeFilter(
                 maxPacketSize)) // drops any packets bigger than the maximum frame size with a warning (accounts for header too)
-            .BatchWeighted(maxFrameSize, tuple => tuple.Item2, tuple => new List<(MqttPacket packet, int readableBytes)>(){ tuple },
+            .BatchWeighted(maxFrameSize, tuple => tuple.Item2.TotalSize, tuple => new List<(MqttPacket packet, PacketSize readableBytes)>(){ tuple },
```

Comment:
This is one place where the https://github.com/petabridge/TurboMqtt/issues/108 errors may have been coming from: we were not including the size of the headers in the batch size calculation, so it's possible that we could have exceeded the maximum frame size in some cases. The fact that we triggered these errors more often with larger packet sizes makes me think this might be true - and also makes me wonder if we do enough logging / validation that we're hitting these maximum frames before we attempt to write them over the network - because in that instance, we might end up leaving off dangling packets that cause the corrupted reads we were observing before.